lib/backend/kubernetes/kubernetes.go

```diff
@@ -277,46 +287,61 @@ func (b *Backend) readSecretData(ctx context.Context, key backend.Key) (*backend
 	}, nil
 }
 
-func (b *Backend) updateSecretContent(ctx context.Context, items ...backend.Item) (*backend.Lease, error) {
+func isConflict(err error) bool { return errors.IsConflict(trace.Unwrap(err)) }
+
+func (b *Backend) updateSecretContent(ctx context.Context, isCreate bool, item backend.Item) (*backend.Lease, error) {
 	// FIXME(tigrato):
 	// for now, the agent is the owner of the secret so it's safe to replace changes
 
-	secret, err := b.getSecret(ctx)
-	if trace.IsNotFound(err) {
-		secret = b.genSecretObject()
-	} else if err != nil {
-		return nil, trace.Wrap(err)
-	}
+	if err := retry.OnError(retry.DefaultRetry, isConflict, func() error {
+		// NOTE: We create the secret in the constructor, so we know it exist at this point.
```

Comment:
Isn't it conceptually simpler to treat a missing secret like an empty one, and create/apply as necessary?

Reply:
@tigrato mentioned in the first pass that Create should not create the secret but only the key and that we should be consistent with sqlite. On the sqlite side, we create the db in the constructor.